inc/class-plugin.php

```diff
@@ -396,6 +397,33 @@ public function filter_sideload_move_temp_file_to_s3( array $file ) {
 		return $file;
 	}
 
+	/**
+	 * Store the attachment filesize in the attachment meta array.
+	 *
+	 * Getting the filesize of an image in S3 involves a remote HEAD request,
+	 * which is a bit slower than a local filesystem operation would be. As a
+	 * result, operations like `wp_prepare_attachments_for_js' take substantially
+	 * longer to complete against s3 uploads than if they were performed with a
+	 * local filesystem.i
+	 *
+	 * Saving the filesize in the attachment metadata when the image is
+	 * uploaded allows core to skip this stat when retrieving and formatting it.
+	 *
+	 * @param array $metadata Attachment metadata.
+	 * @return array Attachment metadata array, with "filesize" value added.
+	 */
+	function set_filesize_in_attachment_meta( $metadata ) {
```

Comment:
```suggestion
	function set_filesize_in_attachment_meta( array $metadata ) : array {
```

Reply:
This project is using Psalm too, so once the unit tests are fixed, I'd imagine we'll need to add the array shape decelerating like elsewhere in the file.